src/test/java/org/flickit/flickitassessmentcore/architecture/dependency/GeneralDependencyArchUnitTest.java

```diff
@@ -0,0 +1,39 @@
+package org.flickit.flickitassessmentcore.architecture.dependency;
+
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+
+import static com.tngtech.archunit.library.Architectures.onionArchitecture;
+import static com.tngtech.archunit.library.DependencyRules.NO_CLASSES_SHOULD_DEPEND_UPPER_PACKAGES;
+import static org.flickit.flickitassessmentcore.architecture.constants.ArchUnitTestConstants.*;
+
+@AnalyzeClasses(packages = {
+    ADAPTER_FULL_PACKAGE,
+    APPLICATION_PORT_IN_FULL_PACKAGE,
+    APPLICATION_PORT_OUT_FULL_PACKAGE,
+    APPLICATION_SERVICE_FULL_PACKAGE,
```

Comment:
درسته اضافه شد و دو تست موجود در این کلاس fail شدن که تست دوم رو درست کردم ولی تست اول داره ایراد میگیره. یکی از خطاهاش اینه که برای مثال کلاس AssessmentReport که در پکیج application.domain.report هست، داخلش کلاس AssessmentColor رو داره که در پکیج بیرونیش یعنی application.domain قرار داره. تست اول میگه نباید اینجوری باشه و AssessmentColor باید داخل ساب پکیج باشه نه پکیج اصلی این دو کلاس. این قاعده برای اینکه بعدا بشه راحت جدا کرد این موارد رو. نظر من اینه که به تست اول نیاز نداریم و میتونیم پاکش کنیم. اگر پاکش نکنیم باید domain modelهامون رو داخل ساب پکیج ببریم.

> A rule that checks that none of the given classes directly depends on classes from upper packages. For an description what "directly depends" means, see JavaClass.getDirectDependenciesFromSelf().
This rule is good practice, because otherwise that might prevent packages on that level from being split into separate artifacts in a clean way in the future.
Example that satisfies the rule:
 mycomponent    |-- api    |       |-- interface MyPublicInterface    |-- subComponentOne    |       |-- class MyPublicInterfaceImplOne implements MyPublicInterface    |-- subComponentTwo            |-- class MyPublicInterfaceImplTwo implements MyPublicInterface
Example that violates the rule:
 mycomponent    |-- interface MyPublicInterface    |-- subComponentOne    |       |-- class MyPublicInterfaceImplOne implements MyPublicInterface // violation    |-- subComponentTwo            |-- class MyPublicInterfaceImplTwo implements MyPublicInterface // violation

Reply:
من فکر می‌کردم منظورش از upper packages، پکیج‌های لایه‌های بالاتر تو معماری شش‌ضلعی یا پیازیه. یعنی فکر می‌کردم چک می‌کنه که لایه‌های داخلی به لایه‌های بیرونی وابسته نباشن. اگر اینطور نیست و با لایه‌ها کاری نداره، به نظرم می‌تونیم حذف کنیم تست رو.